src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts

```diff
@@ -67,10 +74,54 @@ export class ContentComboBox<ITEM_TYPE extends ContentTreeSelectorItem>
         }
 
         this.showAfterReload = false;
-
         this.optionsFactory = new OptionsFactory<ITEM_TYPE>(this.getLoader(), builder.optionDataHelper);
     }
 
+    private createStatusColumn() {
+        this.statusColumn = new GridColumnBuilder()
+            .setId('status')
+            .setName('Status')
+            .setField('displayValue')
+            .setFormatter(ContentRowFormatter.statusSelectorFormatter)
+            .setCssClass('status')
+            .setBoundaryWidth(75, 75)
+            .build();
+    }
+
+    protected initListeners() {
+        const debouncedHandler = AppHelper.debounce(this.handleAvailableSizeChanged.bind(this), 300);
+        ResponsiveManager.onAvailableSizeChanged(this, debouncedHandler);
+    }
+
+    private handleAvailableSizeChanged() {
+        const columns: GridColumn<any>[] = this.getColumnsWithoutCheckbox();
+        const width: number = this.getEl().getWidth();
+
+        if (ResponsiveRanges._360_540.isFitOrSmaller(width)) {
+            if (this.isStatusColumnShown()) {
+                const newColumns: GridColumn<any>[] = columns.filter((column: GridColumn<any>) => column.id !== 'status') ;
+                this.getDataGrid().setColumns(newColumns, true);
+            }
+        } else {
+            if (!this.isStatusColumnShown()) {
+                const newColumns: GridColumn<any>[] = [...columns, this.statusColumn];
+                this.getDataGrid().setColumns(newColumns, true);
+            }
+        }
```

Comment:
I suggest we simplify this a bit.
1. Initially assign `newColumns = columns` (before `if...`)
2. Assign `this.isStatusColumnShown()` to a const (before `if...`). It's called either way, so we might as well just call it once.
3. After `if...` compare lengths of `columns` and `newColumns`. If they are different, call `this.getDataGrid().setColumns()`

Reply:
Reworked in other way : extracted to small functions with self describing names